### ISSUE TITLE
+ operator combines things in a seemingly random order

### ISSUE THREAD

**Issue by clevijoki** [fastbuild/fastbuild#317]
Using this code:

```
.A = [
    .value = 'A'
    .sub = [
        .value = 'A'
        .sub = [
            .value = 'A'
        ]   
    ]   
]

.B = [
    .value = 'B'
    .sub = [
        .value = 'B'
        .sub = [
            .value = 'B'
        ]   
    ]   
]


.AB = .A + .B

Print(.AB)

```

Results in this output:

```
.AB = // Struct
[
    .value = 'AB'
    .sub = // Struct
    [
        .value = 'BA'
        .sub = // Struct
        [
            .value = 'AB'
        ]
    ]
]
```

notice how BA is reversed

**Comment by ffulin** [comment]
Thanks for reporting this. This was actually a very silly bug - so silly I'm a little surprised no-one (including me) noticed it before. I suspect that sub-structures are simply not used by most people, or if they are used, they are used in ways where the order didn't matter.

In any case, I've fixed the bug and added some regression tests (https://github.com/fastbuild/fastbuild/commit/bdb4a0985156431b8da7d4efe02ef403cf82464d). This fix will be in the next release (v0.95). I'll leave this issue open until that new version is released.

**Comment by ffulin** [comment]
v0.95 has now been released and includes this fix for this issue. Thanks again for the bug report!
